StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1137ElementsShouldHaveTheSameIndentation.cs

```diff
@@ -0,0 +1,432 @@
+                break;
+
+            case SyntaxKind.MethodDeclaration:
+            case SyntaxKind.ConstructorDeclaration:
+            case SyntaxKind.DestructorDeclaration:
+            case SyntaxKind.OperatorDeclaration:
+            case SyntaxKind.ConversionOperatorDeclaration:
+                elements.AddRange(((BaseMethodDeclarationSyntax)member).AttributeLists);
+                break;
+
+            case SyntaxKind.GetAccessorDeclaration:
+            case SyntaxKind.SetAccessorDeclaration:
+            case SyntaxKind.AddAccessorDeclaration:
+            case SyntaxKind.RemoveAccessorDeclaration:
+            case SyntaxKind.UnknownAccessorDeclaration:
+                elements.AddRange(((AccessorDeclarationSyntax)member).AttributeLists);
+                break;
+
+            case SyntaxKind.TypeParameter:
+                elements.AddRange(((TypeParameterSyntax)member).AttributeLists);
+                break;
+
+            case SyntaxKind.Parameter:
+                elements.AddRange(((ParameterSyntax)member).AttributeLists);
+                break;
+
+            default:
+                break;
+            }
+
+            elements.Add(member);
+        }
+
+        private static void CheckElements<T>(SyntaxNodeAnalysisContext context, SyntaxList<T> elements)
+            where T : SyntaxNode
+        {
+            if (elements.Count < 2)
+            {
+                return;
+            }
+
+            CheckElements(context, elements.ToImmutableList());
+        }
+
+        private static void CheckElements<T>(SyntaxNodeAnalysisContext context, SeparatedSyntaxList<T> elements)
+            where T : SyntaxNode
+        {
+            if (elements.Count < 2)
+            {
+                return;
+            }
+
+            CheckElements(context, elements.ToImmutableList());
+        }
+
+        private static void CheckElements<T>(SyntaxNodeAnalysisContext context, ImmutableList<T> elements)
+            where T : SyntaxNode
+        {
+            if (elements.Count < 2)
+            {
+                return;
+            }
+
+            elements = elements.RemoveAll(
+                element =>
+                {
+                    SyntaxToken firstToken = GetFirstTokenForAnalysis(element);
+                    return firstToken.IsMissingOrDefault() || !firstToken.IsFirstInLine();
+                });
+
+            if (elements.Count < 2)
+            {
+                return;
+            }
+
+            // Try to reorder the list so the first item is not an attribute list. This element will establish the
+            // expected indentation for the entire collection.
+            int desiredFirst = elements.FindIndex(x => !x.IsKind(SyntaxKind.AttributeList));
+            if (desiredFirst > 0)
+            {
+                T newFirstElement = elements[desiredFirst];
+                elements = elements.RemoveAt(desiredFirst).Insert(0, newFirstElement);
+            }
```

Comment:
If possible, I'd like to wait until we deal with the following items before changing this:

1. Not checking comments and trivia (like directives)
2. Sub-optimal code fix behavior for multi-line elements
3. Issue #2034 (SA1138)